### PR TITLE
support nodegroup as a plugin

### PR DESCRIPTION
**Pull request by qiankunli**
issue #1830 
related pr #2224 
design doc #2227 
`queue.spec.Affinity` is ready by https://github.com/volcano-sh/apis/pull/72  and merged in master branch by #2319, so it is time to push the feature.

### CODE REVIEW

File: pkg/scheduler/framework/arguments.go [hunk 1]
```diff
@@ -85,6 +85,26 @@ func (a Arguments) GetBool(ptr *bool, key string) {
 	*ptr = value
 }
 
+// GetString get the bool value from string
```
Comment (hwdef) [inline pkg/scheduler/framework/arguments.go:88]: The meaning of the comment does not correspond to the function of the function

File: pkg/scheduler/plugins/nodegroup/nodegroup.go [hunk 2]
```diff
@@ -0,0 +1,197 @@
+/*
+Copyright 2019 The Kubernetes Authors.
```
Comment (hwdef) [inline pkg/scheduler/plugins/nodegroup/nodegroup.go:2]:
```suggestion
Copyright 2022 The Volcano Authors.
```

File: pkg/scheduler/framework/arguments.go (hunk 1 above)
Comment (qiankunli) [reply pkg/scheduler/framework/arguments.go]: updated @hwdef

**Comment by william-wang** [comment]
@qiankunli What's the model relationship between queue and nodegroup. Do we allow to configure two queues that affinity the same nodegroup?

**Comment by william-wang** [comment]
@qiankunli please help to add the UT, e2e :)

**Comment by qiankunli** [comment]
> What's the model relationship between queue and nodegroup.

queue and nodegroup are many-to-many relationships

> Do we allow to configure two queues that affinity the same nodegroup?

yes

**Comment by jiamin13579** [comment]
Hi, How long is it available?

**Comment by william-wang** [comment]
@qiankunli please make the ci happy.

**Comment by hwdef** [comment]
still need
/reopen

**Comment by zhoushuke** [comment]
any process?

**Comment by heumsi** [comment]
any process? +1

**Comment by zhaizhch** [comment]
how to calculate the resource in proportion plugin when hard bind queue and node?

**Comment by chenxj1997** [comment]
Does this 1.8.0 version come with this plugin

**Comment by william-wang** [comment]
This PR is replaced with a new PR #3132, so we close it.

**Pull request closed without merging**